src/test/java/ibm/jceplus/junit/base/BaseTestRSASignatureChunkUpdate.java

```diff
@@ -38,7 +39,17 @@ public void setUp() throws Exception {
     public void tearDown() throws Exception {}
 
     public void testSignatureChunks() throws Exception {
-        testSignatureChunkUpdate(1024, "SHA1WithRSA", 100);
+        try {
+            testSignatureChunkUpdate(1024, "SHA1WithRSA", 100);
+        } catch (NoSuchAlgorithmException nsae) {
+            if (providerName.contains("FIPS")) {
+                assertEquals("no such algorithm: SHA1WithRSA for provider OpenJCEPlusFIPS", nsae.getMessage());
+                return;
+            } else {
+                throw nsae;
+            }
+        }
+        
         testSignatureChunkUpdate(1024, "SHA256WithRSA", 100);
         testSignatureChunkUpdate2(1024, "SHA1WithRSA", 100); // Test signing failure with OpenJCEPlusFIPS
```

Comment:
Shouldn't this be removed too?

Reply:
Looking into these paths it appears that this does not currently fail because the underlying method `testSignatureChunkUpdate2` appears to have a FIPS specific check to ensure a error message:

https://github.com/IBM/OpenJCEPlus/blob/050187a432d31b0ca96eaae01922378ff11c8aa6/src/test/java/ibm/jceplus/junit/base/BaseTestRSASignatureChunkUpdate.java#L145C21-L145C22

I would be ok either way although this seems to be ok as is handling the exception in that location.